packages/connect-query/src/utils.ts

```diff
@@ -59,22 +64,27 @@ export const isAbortController = (input: unknown): input is AbortController => {
 /**
  * @see `Updater` from `@tanstack/react-query`
  */
-export type ConnectUpdater<O extends Message<O>> =
-  | PartialMessage<O>
-  | ((prev?: O) => PartialMessage<O> | undefined);
+export type ConnectUpdater<O extends DescMessage> =
+  | MessageInitShape<O>
+  | ((prev?: MessageShape<O>) => MessageInitShape<O> | undefined);
 
 /**
  * This helper makes sure that the Class for the original data is returned, even if what's provided is a partial message or a plain JavaScript object representing the underlying values.
  */
 export const createProtobufSafeUpdater =
-  <I extends Message<I>, O extends Message<O>>(
-    methodSig: Pick<MethodUnaryDescriptor<I, O>, "O">,
+  <I extends DescMessage, O extends DescMessage>(
+    methodSig: Pick<MethodUnaryDescriptor<I, O>, "output">,
     updater: ConnectUpdater<O>,
   ) =>
-  (prev?: O): O => {
+  (prev?: MessageShape<O>): MessageShape<O> => {
     if (typeof updater === "function") {
-      return new methodSig.O(updater(prev));
+      return create(
+        methodSig.output,
+        (
+          updater as (prev?: MessageShape<O>) => MessageInitShape<O> | undefined
```

Comment:
Weird cast, is this even needed?

Reply:
Yes. 

> Unsafe argument of type `any` assigned to a parameter of type `MessageInitShape<DescMessage & O> | undefined`  @typescript-eslint/no-unsafe-argument

Reply:
Ah weird. My IDE didn't have any problems. Good to know.